### PR TITLE
Decouple fetching layer from Redux, part 4

### PR DESCRIPTION
**Pull request by axelboc**
Last two remaining endpoints of the Beamline API: `sendSetAttribute` and `sendExecuteCommand`.

I've also written a tiny bit of developer documentation as you suggested, @fabcor-maxiv. I hope the RST syntax is right, I'm not familiar with it... Would it be feasible to switch to Markdown? :sweat_smile:

### CODE REVIEW

**Comment by fabcor-maxiv** [comment]
> I hope the RST syntax is right, I'm not familiar with it... Would it be feasible to switch to Markdown?

It supports both. It should work with any `.md` file. :)

https://www.sphinx-doc.org/en/master/usage/markdown.html

If not, let me know, I can probably help.

Preview: https://mxcubeweb--1131.org.readthedocs.build/en/1131/dev/front-end.html

Seems fine :)

File: docs/source/dev/front-end.rst
Comment (fabcor-maxiv) [inline docs/source/dev/front-end.rst]: Needs empty line between different nesting levels:
https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#lists-and-quote-like-blocks

```suggestion
- For each back-end API called by the front-end, there is a file under ``src/api/`` named after that API (e.g. ``beamline.js``, ``login.js`` ...)
- For each endpoint, there is a function in the API's front-end file dedicated to making HTTP requests to that endpoint (e.g. ``fetchLoginInfo``, ``sendRunBeamlineAction`` ...)

  - Functions that retrieve resources are named ``fetch<Something>``.
  - Functions that perform actions are named ``send<DoSomething>``.

- Library `wretch <https://github.com/elbywan/wretch>`_ is used for making HTTP requests; it is a thin wrapper around the `Fetch API <https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API>`_.
```

**Review by fabcor-maxiv — DISMISSED** [review]

File: docs/source/dev/front-end.rst
Comment (axelboc) [reply docs/source/dev/front-end.rst]: Oh great, well, if you don't mind, I've switched to Markdown :grin:

Comment (axelboc) [reply docs/source/dev/front-end.rst]: Love the deploy previews of the documentation site, by the way :trophy:

**Comment by axelboc** [comment]
GitHub seems to consider canceled CI jobs as failed. Anyway to change that?

**Comment by fabcor-maxiv** [comment]
> GitHub seems to consider canceled CI jobs as failed. Anyway to change that?

There is some kind of issue about actions workflow being triggered by both the `push` and the `pull request`. I will try to find a fix.

But anyway I do not think it was blocking the review and merge. I had made a review "Changes requested" but then you moved from `.rst` to `.md` so the change is not necessary anymore. I have now dismissed this review that I had made.

**Comment by axelboc** [comment]
> There is some kind of issue about actions workflow being triggered by both the `push` and the `pull request`. I will try to find a fix.

This is how I deal with this in the UI workflow: https://github.com/mxcube/mxcubeweb/blob/develop/.github/workflows/ui.yml#L11

Basically, the `pull_request` trigger is required only for PRs from forks for which the `push` trigger doesn't run, so I skip the `pull_request` trigger when it's not required (i.e. for local PRs)

**Comment by fabcor-maxiv** [comment]
> > There is some kind of issue about actions workflow being triggered by both the `push` and the `pull request`. I will try to find a fix.
> 
> This is how I deal with this in the UI workflow: https://github.com/mxcube/mxcubeweb/blob/develop/.github/workflows/ui.yml#L11
> 
> Basically, the `pull_request` trigger is required only for PRs from forks for which the `push` trigger doesn't run, so I skip the `pull_request` trigger when it's not required (i.e. for local PRs)

As far as I understood this safeguard is not necessary if we keep our pull request branches on our own personal forks instead of on the upstream repository directly. This is a bit strange to me to push branches directly on the upstream repository on GitHub.

**Comment by fabcor-maxiv** [comment]
Let's see if it happens again with this: https://github.com/mxcube/mxcubeweb/pull/1133

**Comment by marcus-oscarsson** [comment]
@fabcor-maxiv #1133 seems to have done the trick

**Comment by axelboc** [comment]
> As far as I understood this safeguard is not necessary if we keep our pull request branches on our own personal forks instead of on the upstream repository directly. This is a bit strange to me to push branches directly on the upstream repository on GitHub.

We are maintainers of the project with write access, so it feels natural to me. :thinking: The simplest process is often the best! :grin: Forks are typically more for external contributors, or when you want to start your own version of a project.

**Comment by fabcor-maxiv** [comment]
>  #1133 seems to have done the trick

Great! :)

---

For the sake of the argument...

In my experience working with forks is more natural especially on forges like GitHub, GitLab, and so on. GitHub is built around this workflow with forks. For me working with the upstream directly is kind of like `sudo` mode. The other advantage of working with forks exclusively is that the upstream repository stays clean with typically only 2 branches `develop` and `main` (and sometimes some long-lived feature branches that need to be shared with all but it is rare).

But I don't mind much either way. In this case I can keep doing pull requests from my own fork.

**Pull request merged**
